src/averageAges.js

```diff
@@ -54,9 +80,22 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
-}
+  const child = people.filter(person =>
+    onlyWithSon
+      ? people.find(mother => mother.name === person.mother)
+        && person.sex === 'm'
+      : people.find(mother => mother.name === person.mother)
+  );
+  const ageDiff = child.map(children => {
+    const childMother = people.find(mother => mother.name === children.mother);
+
+    return children.born - childMother.born;
+  });
```

Comment:
Why can't you use the same `map()` method in `calculateWomenAverageAge` and `calculateMenAverageAge ` ?
This will remove the extra helper. The only one left is the one that gets an array of numbers and returns the average value